wicker/core/datasets.py

```diff
@@ -22,6 +25,72 @@
 FILE_LOCK_TIMEOUT_SECONDS = 300
 
 
+def iterate_bucket_key_chunk_for_size(chunk_list: List[Tuple[str, str]]) -> int:
+    """Iterate on chunk of s3 files to get local length of bytes.
+
+    Args:
+        chunk_list: List of tuples of str, str containing bucket and keys for files on s3 to get byte length.
+
+    Returns:
+        int: total amount of bytes in the file chunk list
+
+    """
+    local_len = 0
+    # create the s3 resource locally and don't pass in. Boto3 docs state to do this in each thread
+    # and not pass around.
+    s3_resource = boto3.resource("s3")
+    for bucket_key_loc in chunk_list:
+        bucket_loc, key_loc = bucket_key_loc
+        # get the byte length for the object
+        byte_length = s3_resource.Object(bucket_loc, key_loc).content_length
+        local_len += byte_length
+    return local_len
+
+
+def get_file_size_s3(input_tuple: Tuple[List[Tuple[str, str]], ValueProxy, Lock]):
+    """Get file size of a list of s3 paths.
+
+    Tuple structure - The tuple contains 5 parts denoted below.
+
+        buckets_keys_chunks_local - The list of tuples denoting bucket and key of files on s3 to
+        parse. Generally column files but will work with any data.
+
+        sum_value - A python shared memory object (ctype) for storing the byte sum across all processes.
+
+        lock - A shared memory lock for locking the sum_value during write procedure.
+
+    Args:
+        input_tuple: A tuple containing the relevant pieces of data used to grab and store byte size.
+            Works across procs so needs a lock and shared memory space.
+
+    Returns:
+        Void function - does all modifications in place on the shared memory value.
+
+    """
+    buckets_keys_chunks_local, sum_value, lock = input_tuple
+
+    # chunk the process again between multiple threads on each proc
+    # done to reduce i/o wait on each individual proc
+    local_chunks = []
+    local_chunk_nums = 500
+    local_chunk_size = len(buckets_keys_chunks_local) // local_chunk_nums
+    for i in range(0, local_chunk_nums):
+        chunk = buckets_keys_chunks_local[i * local_chunk_size : (i + 1) * local_chunk_size]
+        local_chunks.append(chunk)
+
+    # form and add in the last chunk
+    last_chunk_size = len(buckets_keys_chunks_local) - (local_chunk_nums * local_chunk_size)
+    last_chunk = buckets_keys_chunks_local[-last_chunk_size:]
+    local_chunks.append(last_chunk)
```

Comment:
@pickles-bread-and-butter you are missing a check here if the last chunk is empty (i.e. if `len(buckets_keys_chunks_local)` is exactly divisible by `local_chunk_nums`). Perhaps it doesn't really matter if the code handles the possibly-empty chunk. However, it would be a good practice to check if the last chunk is empty before appending it.

Reply:
Done